### PR TITLE
Add separate input renderer

### PR DESCRIPTION
**Pull request by jyash97**
**What this PR includes**

Add Different methods to render inputs
___
**Advantages**
1. Separated methods and state of Dropdown input for searching options to a different file
2. Previously `Sort Field` was not searchable because it was rendered separately, now we can search through sort fields also
3. We can use this components on Editor, and PreviewList as well.
4. We can add tests for checking the consistency of Input values.
___
**What more can be added**
1. We can add `DropdownButton` used on DataField or we can separate the DataField logic to a different module ( Different PR )

### CODE REVIEW

File: components/SearchSandbox/components/InputRenderer.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import React from 'react';
```
Comment (metagrover) [inline components/SearchSandbox/components/InputRenderer.js]: You should move all these inputs in separate individual component files.

Comment (jyash97) [reply components/SearchSandbox/components/InputRenderer.js]: Yeah will make separate components. Is it good to go ? Should I make this changes in other files too ?

**Comment by jyash97** [comment]
@metagrover 
**New Update**

- Add Inputs in `shared` folder
- Introduce `setComponentProps` for setting componentProps
- SearchPreview now uses all the new set of Input Components
___
**How this Input works**

```js
<TextInput
	name="title"
	value="Hey"
	handleChange={value => this.setState({ ...value })}
/>

console.log(this.state.title); // New Updated Value
```

**Comment by jyash97** [comment]
Updated the PR 🚀

**Pull request merged**